config-lib.pl

```diff
@@ -407,5 +407,89 @@ sub save_module_preferences
 	}
 }
 
+# read_config_params(url, insert-prefix)
+# Parses the URL parameters from a URL, and returns a hash ref
+sub read_config_params
+{
+my ($url, $pref) = @_;
+# Extract the part of the URL after the question mark
+my ($query_string) = $url =~ /\?(.*)$/;
+my %params;
+
+# Split the query string on '&' to get the individual key-value pairs
+for my $pair (split(/&/, $query_string)) {
+	my ($key, $value) = split /=/, $pair;
+	$value = &un_urlize($value);
+	my $param_prefix = $pref ? "_cparam-" : "";
+	$params{"$param_prefix$key"} = $value;
+	}
+return \%params;
+}
+
+# print_config_posted_params()
+# Prints hidden fields for all the parameters in the referrer URL
+sub print_config_posted_params
+{
+my $env_referer = $ENV{'HTTP_REFERER'};
+my $params_referer = &read_config_params($env_referer);
+my $params_url = &read_config_params($ENV{'REQUEST_URI'});
```

Comment:
Why do we also need to include the params from the current URL?

Reply:
This is for section config Next and Previous buttons work when Save and Next is clicked.